src/test/kotlin/com/exactpro/th2/common/schema/message/impl/rabbitmq/connection/TestConnectionManager.kt

```diff
@@ -824,29 +816,25 @@ class TestConnectionManager {
         )
 
     @Test
-    @Disabled
-    // TODO: this test is no more relevant
-    // TODO: we need to change test scenario or remove it
     fun `connection manager exclusive queue test`() {
-        RabbitMQContainer(RABBITMQ_IMAGE_NAME)
-            .use { rabbitMQContainer ->
-                rabbitMQContainer.start()
-                LOGGER.info { "Started with port ${rabbitMQContainer.amqpPort}" }
-
-                createConnectionManager(rabbitMQContainer).use { firstManager ->
-                    createConnectionManager(rabbitMQContainer).use { secondManager ->
-                        val queue = firstManager.queueDeclare()
-
-                        assertFailsWith<IOException>("Another connection can subscribe to the $queue queue") {
-                            secondManager.basicConsume(queue, { _, _, _ -> }, {})
-                        }
-
-                        extracted(firstManager, secondManager, queue, 3)
-                        extracted(firstManager, secondManager, queue, 6)
-                    }
+        RabbitMQContainer(RABBITMQ_IMAGE_NAME).use { rabbitMQContainer ->
+            rabbitMQContainer.start()
+            LOGGER.info { "Started with port ${rabbitMQContainer.amqpPort}" }
+
+            createConnectionManager(rabbitMQContainer).use { firstManager ->
+                createConnectionManager(rabbitMQContainer).use { secondManager ->
+                    val queue = firstManager.queueDeclare()
+                    val consumerThread = thread { secondManager.basicConsume(queue, { _, _, _ -> }, {}) }
```

Comment:
The attempt to subscribe to the queue from another connection causes:
_RabbitMQ soft error occupied: (reply-code=405, reply-text=RESOURCE_LOCKED - cannot obtain exclusive access to locked queue 'amq.gen-hdta456EkjDFt7Q_nqsprg' in vhost '/'. It could be originally declared on another connection or the exclusive property value does not match that of the original declaration., class-id=60, method-id=20) on channel AMQChannel(amqp://guest@127.0.0.1:32775/,1)_ error.

From my point of view, this looks like an unrecoverable error. According to [official documentation](https://www.rabbitmq.com/queues.html#exclusive-queues) the exclusive queue can only be managed by the connection it was created by. We must propagate the error back to the consumer.

@Nikita-Smirnov-Exactpro Do you have any objection to this?

Reply:
Common generates unique name for exclusive queue. The error above is critical for the current implementation.
Also, some messages can be missed if we recreate exclusive queue with the same name.
The problem should be critical as result

Reply:
Critical error in this case looks like a concept problem and we should think about solving in future